tlatools/org.lamport.tlatools/javacc/tla+.jj

```diff
@@ -2304,72 +2218,6 @@ expecting = "Expression or Op. Symbol"; }
   { epa(); zn[2] = tn; return new SyntaxTreeNode(mn, N_Substitution, zn ); }
 }
 
-/***************************************************************************
-* Substitution ::=                                                         *
-*     ( Identifier | NonExpPrefixOp | InfixOp | PostfixOp)                 *
-*     <SUBSTITUTE> ( <op_76> | Lambda | Expression )                       *
-*                                                                          *
-* Note: <op_76> is "-.", the prefix - operator.                            *
-*       <SUBSTITUTE> is "<-"                                               *
-*                                                                          *
-* Modified 27 March 2007 by LL to allow Lambda substitutions.              *
-***************************************************************************/
-SyntaxTreeNode
-OldSubstitution() : {
```

Comment:
This is mentioned in `tla+.html`. It isn't used there either.

Is that file autogenerated? It looks handwritten. If it's handwritten, let's remove this production from it by hand, as well as any others removed by this PR.

Reply:
I believe tla+.html is auto-generated by JavaCC. It may have been subsequently modified by hand. I will have to look into how to generate it to keep it up to date.

Reply:
I figured out how to do it and added another commit in this PR adding the functionality to the `generate` ant task.

Reply:
I split this out into this PR: https://github.com/tlaplus/tlaplus/pull/1043

This does mean that if this or that PR is merged first, the other will have to be updated or else the CI will fail when both are merged.